cmake/QtFunctions.cmake

```
@@ -20,40 +20,41 @@ macro(find_qt_package)
         set(MY_OPTIONS_COMPONENTS "")
     endif()
 
-    find_package(Qt6 COMPONENTS ${MY_OPTIONS_COMPONENTS} QUIET)
-    if(NOT Qt6_FOUND)
-        if(${MY_OPTIONS_REQUIRED})
-            find_package(Qt5 5.15 COMPONENTS ${MY_OPTIONS_COMPONENTS} REQUIRED)
-        else()
-            find_package(Qt5 5.15 COMPONENTS ${MY_OPTIONS_COMPONENTS})
+    if(${MY_OPTIONS_REQUIRED})
+        set(QT_SEARCH_MODE REQUIRED)
+    else()
+        set(QT_SEARCH_MODE QUIET)
+    endif()
+
+    if(QT_DEFAULT_MAJOR_VERSION STREQUAL "6")
+        find_package(Qt6 COMPONENTS ${MY_OPTIONS_COMPONENTS} ${QT_SEARCH_MODE})
+    elseif(QT_DEFAULT_MAJOR_VERSION STREQUAL "5")
+        find_package(Qt5 5.15 COMPONENTS ${MY_OPTIONS_COMPONENTS} ${QT_SEARCH_MODE})
+    else() # QT_DEFAULT_MMAJOR_VERSION not set, first search 6, then 5.
+        find_package(Qt6 COMPONENTS ${MY_OPTIONS_COMPONENTS})
```

Don't we need to have this one `QUIET` to avoid failure and allow to go to 5 ?
```suggestion
        find_package(Qt6 COMPONENTS ${MY_OPTIONS_COMPONENTS} QUIET)
```